packages/cape/src/methods.ts

```diff
@@ -212,6 +178,98 @@ export abstract class Methods {
       throw e;
     }
   }
+
+  /**
+   * Retrieve a Cape key using your authentication token or function token. This method will manage the entire lifecycle for you.
+   * The returned key is stored in the `client` object as a parameter for encrypt.
+   *
+   * @example
+   * ```ts
+   * const client = new Cape({ authToken: 'my-auth-token' });
+   * await key = client.key();
+   * ```
+   */
+  public async key(): Promise<string> {
+    try {
+      const path = this.getCanonicalPath(`/v1/key`);
+
+      const attestationUserData = await this.connect_(path);
+      const obj = JSON.parse(attestationUserData);
+      const userData = obj.key;
+      const keyString = '-----BEGIN PUBLIC KEY-----\n' + addNewLines(userData) + '\n-----END PUBLIC KEY-----';
+      this.encryptKey = new TextEncoder().encode(keyString);
+      return keyString;
+    } finally {
+      this.disconnect();
+    }
+  }
+
+  /**
+   * Encrypt the
+   * The returned key is stored in the `client` object
+   *
+   * @example
+   * ```ts
+   * const myInput = <YOUR_INPUT>;
+   * const client = new Cape({ authToken: 'my-auth-token' });
+   * await key = client.key();
+   * const encrypted = client.encrypt(myInput)
+   *
+   * ```
+   */
+  public async encrypt(input: string): Promise<string> {
+    if (this.encryptKey == null) {
+      await this.key();
+    }
+    const bytesToEncrypt = new TextEncoder().encode(input);
+    const key = this.getEncryptKey() || new Uint8Array();
```

Comment:
Should there really be a default empty case here?

Reply:
The key could be undefined upon defining the Cape object. I would argue that this is not the default case for encrypt, because in the block on top, if the key doesn't exist, we retrieve it first from an enclave. This is more to get around the type limitation. Maybe a better way is to check if the value is null and explicitly throw an error if key doesn't exist.

Reply:
@eric-capeprivacy :+1: